File: Leetcode/SortColors.rs
// https://leetcode.com/problems/sort-colors

struct Solution {}

impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        let mut cnt = vec![0; 3];
        for &x in nums.iter() {
            cnt[x as usize] += 1;
        }
        for x in nums.iter_mut() {
            for t in 0..3 {
                if cnt[t] > 0 {
                    cnt[t] -= 1;
                    *x = t as i32;
                    break;
                }
            }
        }
    }
}

File: Leetcode/SortColors.c
// https://leetcode.com/problems/sort-colors

void sortColors(int* nums, int numsSize) {
    int cnt[3] = {0};

    for (int i = 0; i < numsSize; ++i) {
        cnt[nums[i]] += 1;
    }

    for (int el = 0, i = 0; el < 3; ++el) {
        while (cnt[el]--) {
            nums[i++] = el;
        }
    }
}